core/src/main/java/org/bitcoinj/script/Script.java

```diff
@@ -117,12 +118,38 @@ private Script() {
      * @param programBytes Array of program bytes from a transaction.
      */
     public Script(byte[] programBytes) throws ScriptException {
-        program = programBytes;
-        parse(programBytes);
-        creationTimeSeconds = 0;
+        this(ScriptPattern.BITCOIN, programBytes);
     }
 
     public Script(byte[] programBytes, long creationTimeSeconds) throws ScriptException {
+        this(ScriptPattern.BITCOIN, programBytes, creationTimeSeconds);
+    }
+
+    /** Creates an empty script that serializes to nothing. */
+    private Script(ScriptPattern pattern) {
+        this.pattern = pattern;
+        chunks = Lists.newArrayList();
+    }
+
+    // Used from ScriptBuilder.
+    Script(ScriptPattern pattern, List<ScriptChunk> chunks) {
+        this.pattern = pattern;
+        this.chunks = Collections.unmodifiableList(new ArrayList<>(chunks));
+        creationTimeSeconds = Utils.currentTimeSeconds();
+    }
+
+    /**
+     * Construct a Script that copies and wraps the programBytes array. The array is parsed and checked for syntactic
+     * validity.
+     * @param pattern The pattern used form methods like isSentToAddress
+     * @param programBytes Array of program bytes from a transaction.
+     */
+    public Script(ScriptPattern pattern, byte[] programBytes) throws ScriptException {
+        this(pattern, programBytes, 0);
+    }
+
+    public Script(ScriptPattern pattern, byte[] programBytes, long creationTimeSeconds) throws ScriptException {
```

Comment:
Maybe add a JavaDoc here, too? (Yes, it's very similar to the above.)

Reply:
Added